### ISSUE TITLE
finetune on other domain, the validation is a bit noisy

### ISSUE THREAD

**Issue by jingyangcarl** [prs-eth/Marigold#61]
Awesome work!! I'm trying to use the same fine-tuned protocol over another domain.

However, I suffered from the noisy results. The training lasts for 2 days on A100.

Is there any chance I can get some insights to improve the results?

Best

**Comment by markkua** [comment]
Can you give more details?

**Comment by jingyangcarl** [comment]
Hi @markkua thanks for the message. I modified my training code from [instruct_pix2pix](https://github.com/huggingface/diffusers/blob/main/examples/instruct_pix2pix/train_instruct_pix2pix.py) where I changed the input and output conv layer of the unet to match my task, e.g. I'm taking RGB images as input and output albedo (also in RGB). I also modified the Marigold pipeline to adapt the inference to enable the three-channel output. 

Also, I'm using DDPM during training with 1k diffusion steps and DDIM with 50 sample steps at inference. I also set the ensemble to 10 to match your implementation details. The training takes roughly 48 hrs on A100 and I attached an example below.

Input: 
![image](https://github.com/prs-eth/Marigold/assets/24415835/e6adbd7c-6f47-4848-a12d-16228c382d7a)

Output:
![image](https://github.com/prs-eth/Marigold/assets/24415835/4037dd74-6cce-4024-8e1e-4df25026a793)

I think the training works since it generates the albedo, however, it's noticeable that there is still noise on the centered object. The generation is not clean enough.

Looking forward to hearing your insights. 

Best,
Jing

**Comment by markkua** [comment]
Hi, 

From your description, I assume that you adjust the unet output layer to have three latents. As you are exporting three-channel albedo, have you tried directly using one latent, which gives you three channels after decoding.

On the other hand, adapting Marigold to a different modality might require different training hyperparameters.

Best.

**Comment by jingyangcarl** [comment]
Hi @markkua ,

Thanks for the message. I'm currently outputting only one latent code (4-dim) as the output of the unet. I believe the network structure has not changed that much. The colored output is also via a similar decoder in [marigold_pipeline](https://github.com/prs-eth/Marigold/blob/be3065260f19b02a807c93b572a6d10ab396bd57/marigold/marigold_pipeline.py) but comment [this line](https://github.com/prs-eth/Marigold/blob/be3065260f19b02a807c93b572a6d10ab396bd57/marigold/marigold_pipeline.py#L439).

As for the training, I also prepared 10k image pairs of input RGB images and albedo images.

Please let me know if I missed anything or misunderstood anything.

Best,
Jing

**Comment by jingyangcarl** [comment]
Solved, thx

**Comment by 2hiTee** [comment]
> Solved, thx

Hi! I also met the same problem, which I adopt the same training strategy for another domain, and I get worse results as well. I would be appreciate if you could provide some training strategies about fintuning SD for another domain. Thanks again!
